collections_hierarchy/main_collections.py

```diff
@@ -9,6 +9,10 @@ class List(ComparableMixin,
            IndexableMixin,
            AppendableMixin):
     DATA_DEFAULT_INITIAL = []
+    
+    def __init__(self, values):
+        self.values = values
+        self.i = -1
```

Comment:
You could add the option to initialize without any parameters. You had that option in the ConstructibleMixin `init` but it gets overwritten here. 

Also if you initialized `i` in the `iter` Mixin you wouldn't have to deal with it in these children classes which might be nice/